Client/Frontend/Passcode/PasscodeViewController.swift

```diff
@@ -0,0 +1,135 @@
+            $0.backgroundColor = .braveBlurple
+            $0.isHidden = true
+        }
+
+        override func viewDidLoad() {
+            super.viewDidLoad()
+
+            view.addSubview(backgroundView)
+            view.addSubview(lockImageView)
+            view.addSubview(unlockButton)
+            backgroundView.snp.makeConstraints {
+                $0.edges.equalTo(view)
+            }
+            lockImageView.snp.makeConstraints {
+                $0.center.equalTo(view)
+            }
+            unlockButton.snp.makeConstraints {
+                $0.leading.greaterThanOrEqualToSuperview().offset(20)
+                $0.trailing.lessThanOrEqualToSuperview().offset(20)
+                $0.centerX.equalToSuperview()
+                $0.height.greaterThanOrEqualTo(44)
+                $0.width.greaterThanOrEqualTo(230)
+                $0.top.equalTo(lockImageView.snp.bottom).offset(60)
+            }
+        }
+    }
+    
+    private let lockedViewController = LockedViewController()
+    
+    private var cancellables: Set<AnyCancellable> = []
+    private var protectedWindow: UIWindow
+    private var passcodeWindow: UIWindow
+    private let context = LAContext()
+    
+    private var isVisible: Bool = false {
+        didSet {
+            passcodeWindow.isHidden = !isVisible
+            if isVisible {
+                passcodeWindow.makeKeyAndVisible()
+            } else {
+                protectedWindow.makeKeyAndVisible()
+            }
+        }
+    }
+    
+    init?(window: UIWindow) {
+        guard let scene = window.windowScene else { return nil }
+        protectedWindow = window
+        
+        passcodeWindow = UIWindow(windowScene: scene)
+        passcodeWindow.windowLevel = .init(UIWindow.Level.statusBar.rawValue + 1)
+        passcodeWindow.rootViewController = lockedViewController
+        
+        lockedViewController.unlockButton.addTarget(self, action: #selector(tappedUnlock), for: .touchUpInside)
+
+        NotificationCenter.default.publisher(for: UIApplication.didEnterBackgroundNotification)
+            .sink(receiveValue: { [weak self] _ in
+                guard let self = self else { return }
+                // Update visibility when entering background
+                self.isVisible = Preferences.Privacy.lockWithPasscode.value &&
+                    self.context.canEvaluatePolicy(.deviceOwnerAuthentication, error: nil)
+            })
+            .store(in: &cancellables)
+        
+        NotificationCenter.default.publisher(for: UIApplication.willEnterForegroundNotification)
+            .merge(with: NotificationCenter.default.publisher(for: UIApplication.didFinishLaunchingNotification))
+            .sink(receiveValue: { [weak self] _ in
+                guard let self = self else { return }
+                let isLocked = Preferences.Privacy.lockWithPasscode.value &&
+                    self.context.canEvaluatePolicy(.deviceOwnerAuthentication, error: nil)
```

Comment:
See https://github.com/brave/brave-ios/pull/3922#discussion_r669117205 

But yes, if the user disables their PIN we will let them access the app since we can no longer evaluate the policy. Do we also need to create a new `LAContext` each time?

Reply:
We don't need a new context each time, but we do need one if it somehow got invalidated. The state of the context can change when you go from background to foreground.

Reply:
Went with create new `LAContext` each time we foreground the app and checked explicitly for `passcodeNotSet` error code